src/tfs/tfs.c

```diff
@@ -429,64 +448,32 @@ static void add_extent_to_file(fsfile f, extent ex)
     table_set(e, sym(length), value_from_u64(h, range_span(ex->node.r)));
     table_set(e, sym(allocated), value_from_u64(h, ex->allocated));
     if (ex->uninited)
-        table_set(e, sym(uninited), null_value());
+        table_set(e, sym(uninited), null_value);
+    symbol offs = intern_u64(ex->node.r.start);
+    fs_status s = filesystem_write_eav(f->fs, extents, offs, e);
+    if (s != FS_STATUS_OK) {
+        destruct_tuple(e, true);
```

Comment:
At this point we cannot destroy the extents tuple (even if empty) because it has been encoded in the log already. We might remove a reference to it from the file metadata (as in `table_set(f->md, sym(extents), 0);`), but I think it's better to keep a reference to it, otherwise if later a new attempt is made to add an extent to the file, the extents tuple would be wastefully re-allocated.